packages/e2e-tests/test/test-shell.ts

```diff
@@ -269,6 +249,64 @@ export class TestShell {
     ]);
   }
 
+  /**
+   * Like the `eventually` utility, but instead of calling the callback on a timer,
+   * the callback is called as output is emitted.
+   */
+  eventually<T = unknown>(
```

Comment:
I might need help to pick a better name for it 🤔 The reason I choose this, is because it acts as a replacement for the pull-based `eventually` utility used elsewhere.